src/provider/types.ts

```diff
@@ -0,0 +1,213 @@
+import { EventEmitter } from 'events'
```

Comment:
Where do all this type definitions come from?
I assume you didn't make them by hand. Did you use any tool or are they copied from somewhere else?

Reply:
From node, it's the default EventEmitter that everyone uses

Reply:
I was meaning the typescript ones that follows, but I know where they came from. resolving...